include/crm/common/messages_internal.h

```diff
@@ -15,6 +15,21 @@
 #include <crm/common/ipc_internal.h>        // pcmk__client_t
 #include <crm/common/results_internal.h>    // pcmk__action_result_t
 
+enum pcmk__request_flags {
+    pcmk__request_none          = UINT32_C(0),
+
+    /* It would be nice if we could check for synchronous requests generically,
+     * but each daemon uses its own call options, so the daemons are responsible
+     * for setting this flag when appropriate.
+     */
+    pcmk__request_sync          = (UINT32_C(1) << 0),
+
```

Comment:
We might introduce a sub-structure that holds daemons specifics like the command attribute and a mask for the sync-flag.

Reply:
Yep that's the direction I want to go in, but I want to convert a few daemons before coming up with a design since they're all so different